### PR TITLE
【Hackathon 4th No.25】为 Paddle 新增 paddle.sparse.any 稀疏 API

### PR DESCRIPTION
**Pull request by zrr1999**
任务说明 https://github.com/PaddlePaddle/community/blob/master/hackthon_4th/%E3%80%90PaddlePaddle%20Hackathon%204%E3%80%91%20%E6%A0%B8%E5%BF%83%E6%A1%86%E6%9E%B6%E5%BC%80%E6%BA%90%E8%B4%A1%E7%8C%AE%20API%20%E5%BC%80%E5%8F%91%E4%BB%BB%E5%8A%A1%E5%90%88%E9%9B%86.md#task25

由于在实现上与sparse.sum相近，因此rfc也主要根据 https://github.com/PaddlePaddle/community/pull/411 修改

### CODE REVIEW

**Comment by zrr1999** [comment]
any 算子似乎并没有比较合理的梯度定义，在 Dense 张量上似乎也没有实现反向传播

File: rfcs/APIs/20230405_api_design_for_sparse_any.md [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+另外当 axis=None 时所有元素进行 any 运算。
+
+## 3、意义
+支持稀疏 tensor 的 any 操作，丰富基础功能，提升稀疏 tensor 的 API 完整度。
+
+# 二、飞桨现状
+目前paddle缺少相关功能实现。
+
+# 三、业内方案调研
+## Pytorch
```
Comment (zkh2016) [inline rfcs/APIs/20230405_api_design_for_sparse_any.md:25]: pytorch的any操作是否支持sparse tensor？

File: rfcs/APIs/20230405_api_design_for_sparse_any.md [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+(const Tensor& self, int64_t dim, bool keepdim, const Tensor& result) {
+  allany_impl<0>(self, result, dim, keepdim, or_stub);
+}
+```
+
+## scipy
+scipy.sparse库中没有any()函数。但是，可以使用numpy库中的any()函数在稀疏矩阵中执行相同的操作。
+
+## paddle DenseTensor
+DenseTensor中的sum被定义为paddle.any(x, axis=None, dtype=None, keepdim=False, name=None)，
```
Comment (zkh2016) [inline rfcs/APIs/20230405_api_design_for_sparse_any.md:65]: sum->any，包括后面的sum也应该相应修改下。

File: rfcs/APIs/20230405_api_design_for_sparse_any.md (hunk 1 above)
Comment (zrr1999) [reply rfcs/APIs/20230405_api_design_for_sparse_any.md:25]: 根据文档的描述，torch.any支持稀疏矩阵和稠密矩阵，两种实现是写在一起的
image
一些其他算子，例如sum是单独实现的。

Comment (zrr1999) [reply rfcs/APIs/20230405_api_design_for_sparse_any.md:25]: torch应该只支持coo格式，我测试了一下，csr确实是不支持的

```python
import torch
import torch.nn as nn

x = torch.Tensor([[[1, 0], [1, 0]], [[1, 0], [1, 0]], [[1, 0], [1, 0]]])
coo_x = x.to_sparse_coo()
csr_x = x.to_sparse_csr()

print(torch.any(coo_x))
print(torch.any(csr_x))


```

File: rfcs/APIs/20230405_api_design_for_sparse_any.md (hunk 2 above)
Comment (zrr1999) [reply rfcs/APIs/20230405_api_design_for_sparse_any.md]: 好的，已修改
